app/models/vm.rb

```diff
@@ -1,6 +1,7 @@
 class Vm < VmOrTemplate
   default_scope { where(:template => false) }
-
+  has_one :container_deployment, :through => :container_deployment_node
```

Comment:
> I don't think so, every vm can serve as a node/master in one deployment of OpenShift.

ok, but be aware, that this is not enforced by the DB, so rails will just return the first record that matches.
In case there is a way to encode this semantic in the DB it would be better